### PR TITLE
Add public and private to the spec

### PR DESCRIPTION
**Pull request by lydia-duncan**
This change specifies the behavior of modules, variables, functions,
iterators, and type aliases when declared to be public or private.  In
the sections for records and classes, it lists the behavior as
unspecified until we have a discussion on the topic (since inheritance
makes the story more tricky than the other cases).  I have not updated
the section for unions, though it should not be tricky like the case of
classes and records.

While here, I altered a reference to "type-specifier" in the type alias
section to match that of similar references to the syntax declarations.
I also moved the majority of the Use Statement section under the
renamed Visibility section in the Modules chapter, as it seemed logical
to keep the different ways of accessing a module's symbols together.

### CODE REVIEW

File: spec/Functions.tex [hunk 1]
```diff
@@ -227,6 +231,11 @@ \section{Procedure Definitions}
 Operator overloading is supported on the operators listed
 above (see \sntx{operator-name}).
 
+The optional \sntx{privacy-specifier} keywords indicate the visibility
+of module level functions to outside modules.  By default, functions are
+publicly visible.  More details on visibility can be found in
+~\rsec{Visibility}.
+
```
Comment (vasslitvinov) [inline spec/Functions.tex]: Above, replace "functions" with "procedures".

File: spec/Functions.tex [hunk 2]
```diff
@@ -1076,9 +1085,9 @@ \subsection{Determining Visible Functions}
 function} if the name of the function is the same as the name of the
 function call and the function is defined in the same scope as the
 function call or a lexical outer scope of the function call, or if the
-function is defined in a module that is used from the same scope as
-the function call or a lexical outer scope of the function call.
-Function visibility in generic functions is discussed
+function is publicly defined in a module that is used from the same
```
Comment (vasslitvinov) [inline spec/Functions.tex]: I suggest "function has public visibility and is defined in a module ..."

Rationale: we do not have a definition for "publicly defined".

File: spec/Modules.tex [hunk 3]
```diff
@@ -9,9 +9,10 @@
 Module definitions are described in~\rsec{Module_Definitions}.  The
 relation between files and modules is described
 in~\rsec{Implicit_Modules}.  Nested modules are described
-in~\rsec{Nested_Modules}.  Module uses and explicit naming of symbols
-are described in~\rsec{Using_Modules}.  The execution of a program and
-module initialization is described in~\rsec{Program_Execution}.
+in~\rsec{Nested_Modules}.  The visibility of symbols within a module
+in relation to other modules is described in~\rsec{Visibility}.  The
+execution of a program and module initialization is described
```
Comment (vasslitvinov) [inline spec/Modules.tex]: I'd like it to say "The visibility of a module's symbols by users of the module is described in~\rsec{Visibility}."

File: spec/Modules.tex [hunk 4]
```diff
@@ -21,7 +22,11 @@ \section{Module Definitions}
 A module is declared with the following syntax:
 \begin{syntax}
 module-declaration-statement:
-  `module' module-identifier block-statement
+  privacy-specifier[OPT] `module' module-identifier block-statement
+
+privacy-specifier:
+  `private'
+  `public'
```
Comment (vasslitvinov) [inline spec/Modules.tex]: Hm shouldn't 'privacy-specifier' be declared only once? This duplicates the rule in the Procedures chapter.

Comment (lydia-duncan) [inline spec/Modules.tex]: I had thought the intention was for the syntax sections to be distinct and so saw no problem with the duplication.  I don't think someone should have to go searching for the definition of a rule if it is immediately relevant.

File: spec/Functions.tex (hunk 2 above)
Comment (lydia-duncan) [inline spec/Functions.tex]: Is "publicly declared" sufficient?

**Comment by lydia-duncan** [comment]
@vasslitvinov, the changes are pushed

File: spec/Modules.tex [hunk 5]
```diff
@@ -182,14 +188,38 @@ \section{Nested Modules}
...
+\label{Access_Of_Module_Contents}
+\index{modules!access}
+
+A module can be used by code outside of that module depending on the
+visibility of the module itself~(\rsec{Visibility_Of_A_Module}).  This
+allows access to its visible contents~(\rsec{Visibility_Of_Symbols})
+without the need for explicit naming~(\rsec{Explicit_Naming}).  Using
+modules is accomplished via the use statement as defined
+in~\rsec{Using_Modules}.
+
```
Comment (vasslitvinov) [inline spec/Modules.tex]: I would like the above paragraph to read like this: 1st sentence unchanged, then:

This
allows access to its visible contents~(\rsec{Visibility_Of_Symbols})
via explicit naming~(\rsec{Explicit_Naming})
or the use statement~(\rsec{Using_Modules}).

**Pull request merged**